packages/ipfs-message-port-client/README.md

```diff
@@ -0,0 +1,141 @@
+[![Codecov branch](https://img.shields.io/codecov/c/github/ipfs/js-ipfs/master.svg?style=flat-square)](https://codecov.io/gh/ipfs/js-ipfs)
+[![Dependency Status](https://david-dm.org/ipfs/js-ipfs/status.svg?path=packages/ipfs-message-port-client)](https://david-dm.org/ipfs/js-ipfs?path=packages/ipfs-message-port-client)
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/feross/standard)
+
+> A client library for the IPFS API over [message channel][]. This client library provides (subset) of [IPFS API](https://github.com/ipfs/js-ipfs/tree/master/docs/api) enabling applications to work with js-ipfs running in the different JS e.g. [SharedWorker][].
+
+
+## Lead Maintainer <!-- omit in toc -->
+
+[Alex Potsides](https://github.com/achingbrain)
+
+## Table of Contentens <!-- omit in toc -->
+
+- [Install](#install)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Install
+
+```bash
+$ npm install --save ipfs-message-port-client
+```
+
+## Usage
+
+This client library works with IPFS node over the [message channel][] and assumes that IPFS node is provided via `ipfs-message-port-server` on the other end.
+
+It provides following API subseset:
+
+- [`ipfs.dag`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/DAG.md)
+- [`ipfs.block`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/BLOCK.md)
+- [`ipfs.add`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/FILES.md#ipfsadddata-options)
+- [`ipfs.cat`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/FILES.md#ipfscatipfspath-options)
+- [`ipfs.files.stat`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/FILES.md#ipfsfilesstatpath-options)
+
+Client can be instantiated from the [`MessagePort`][] instance
+
+
+```js
+const IPFSClient = require('ipfs-message-port-client')
+
+
+const main = async () => {
+  const worker = new SharedWorker(IPFS_SERVER_URL)
+  const ipfs = IPFSClient.from(worker.port)
+  const data = ipfs.cat('/ipfs/QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n')
+
+  for await (const chunk of data) {
+    console.log(chunk)
+  }
+}
+```
+
+It is also possible to instantiate detached client, which can be attach it to
+the server later on. This is useful when server port is received via message
+from other JS context (e.g. iframe)
+
+> Note: Client will queue all API calls and only execute them once it is
+> attached (unless they timeout or are aborted in the meantime).
+
+```js
+const IPFSClient = require('ipfs-message-port-client')
+
+
+const ipfs = IPFSClient.detached()
+
+const main = async () => {
+  const data = ipfs.cat('/ipfs/QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n')
+
+  for await (const chunk of data) {
+    console.log(chunk)
+  }
+}
+
+window.onload = main
+window.onmessage = ({ports}) => {
+  IPFSClient.attach(ports[0])
+}
+```
+
+### Additional Options
```

Comment:
Can we rename this section to not bury the lead, and add it to TOC at the top for easy discovery? 
```suggestion
### Notes on Performance
```